app/pages/onboarding/03-restore-wallet/restore-wallet.tsx

```diff
@@ -23,18 +23,22 @@ export const RestoreWallet: React.FC = () => {
   useBackButton(routes.WELCOME);
 
   const [mnemonic, setMnemonic] = useState('');
+  const [hasSubmitted, setHasSubmitted] = useState(false);
   const [error, setError] = useState<string | null>(null);
   const history = useHistory();
   const dispatch = useDispatch();
 
   const handleMnemonicInput = (e: React.FormEvent<HTMLInputElement>) => {
+    setHasSubmitted(false);
+    setError(null);
```

Comment:
I think we should check these and only set them if they are not the values we are setting. Ideally we'd prob want to debounce this setMnemonic fn, too.